### PR TITLE
setup mailer for donors and fulfill option for requestor

### PR DESCRIPTION
**Pull request by jaspreet21anand**

### CODE REVIEW

File: app/controllers/donor_requirements_controller.rb [hunk 1]
```diff
@@ -2,6 +2,7 @@ class DonorRequirementsController < ApplicationController
 
   before_action :set_requirement, only: [:create, :destroy, :mark_donated]
   before_action :set_donor_requirement, only: [:destroy, :mark_donated]
+  before_action :check_requirement_status, only: [:destroy, :mark_donated]
```
Comment (ShubhamGupta) [inline app/controllers/donor_requirements_controller.rb]: Should be a callback.

File: app/controllers/requirements_controller.rb [hunk 2]
```diff
@@ -65,7 +69,11 @@ def destroy
   end
 
   def fulfill
-    flash.now[:alert] = 'This request has no donors, you can probably delete it.' unless @requirement.fulfill!
+    if @requirement.fulfill!
```
Comment (ShubhamGupta) [inline app/controllers/requirements_controller.rb:72]: Only owner should be able to fulfill.

File: app/mailers/donor_mailer.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+class DonorMailer < ActionMailer::Base
+  default from: 'mail2freeka@gmail.com'
+
+  def thank_interested_donor(donor, requirement)
+    @donor = donor
+    @requirement = requirement
+    mail to:@donor.email, subject:'Thanks for showing Interest'
```
Comment (ShubhamGupta) [inline app/mailers/donor_mailer.rb]: Proper spacing.

File: app/mailers/donor_mailer.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  end
+
+  def thank_fulfilling_donor(donor, requirement)
+    @donor = donor
+    @requirement = requirement
+    mail to:@donor.email, subject:'Whole hearted thanks for your donation'
```
Comment (ShubhamGupta) [inline app/mailers/donor_mailer.rb]: Proper spacing.

File: app/views/devise/mailer/confirmation_instructions.html.erb [hunk 5]
```diff
@@ -0,0 +1,5 @@
+<p>Welcome <%= @resource.name %>!</p>
```
Comment (ShubhamGupta) [inline app/views/devise/mailer/confirmation_instructions.html.erb:1]: Why are these devise mailers generated now?

File: app/views/requirements/show.html.erb [hunk 6]
```diff
@@ -39,15 +39,23 @@
           </div>
           <div class='pull-right'>
             <% @donor_requirement = @requirement.donor_requirements.find_by(donor_id: current_user.try(:id)) %>
-            <% if @donor_requirement %>
-              <% if @donor_requirement.donated? %>
-                <div> You have already donated this item. </div>
+            <% if @requirement.fulfilled? %>
```
Comment (ShubhamGupta) [inline app/views/requirements/show.html.erb:42]: Check these conditions once again.

File: app/views/devise/mailer/confirmation_instructions.html.erb (hunk 5 above)
Comment (jaspreet21anand) [inline app/views/devise/mailer/confirmation_instructions.html.erb:1]: They pointed out to welcome the user by its name, which was earlier by email. So changed it too.

File: app/views/requirements/show.html.erb (hunk 6 above)
Comment (jaspreet21anand) [inline app/views/requirements/show.html.erb:42]: Checked. Seems good.

**Pull request merged**